users/serializers.py

```diff
@@ -0,0 +1,73 @@
+import django.contrib.auth.password_validation as password_validation
+from django.contrib.auth.models import User
+from rest_framework import serializers
+from rest_framework.authtoken.models import Token
+
+
+class UserDetailSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username',)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username', 'password')
+
+    def create(self, data):
+        user = User(username=data["username"])
+        user.set_password(data["password"])
+
+        if "email" in data:
+            user.email = data["email"]
+
+        user.save()
+        Token.objects.create(user=user)
+        return user
+
+    def validate_password(self, value):
+        password_validation.validate_password(value)
+        return value
+
+
+class LoginSerializer(serializers.Serializer):
+    username = serializers.CharField(required=True)
+    password = serializers.CharField(required=True)
+
+    def validate(self, data):
+        try:
+            user = User.objects.get(username=data["username"])
+        except Exception as e:
+            raise serializers.ValidationError(detail={"username": e.args[0]})
```

Comment:
You should return meaningful error messages in the API response as you did for password field.